morpheus/llm/services/openai_chat_service.py

```diff
@@ -72,27 +110,46 @@ def get_input_names(self) -> list[str]:
 
         return input_names
 
-    def _create_messages(self, prompt: str, assistant: str = None) -> list[dict[str, str]]:
-        messages = [
-            {
-                "role": "system", "content": "You are a helpful assistant."
-            },
-            {
-                "role": "user", "content": prompt
-            },
-        ]
+    @contextmanager
+    def _api_logger(self, inputs: typing.Any):
 
-        if (self._set_assistant):
+        message_id = self._parent._get_message_id()
+        start_time = time.time()
+
+        api_logger = _ApiLogger(message_id=message_id, inputs=inputs)
+
+        yield api_logger
+
+        end_time = time.time()
+        duration_ms = (end_time - start_time) * 1000.0
+
+        log_str = dedent("""
+        ============= MESSAGE %d START ==============
+                        --- Input ---
+        %s
+                        --- Output --- (%f ms)
+        %s
+        =============  MESSAGE %d END ==============
+        """).strip("\n")
```

Comment:
I would be tempted to make this a class property or a global since its a constant, and would avoid the `dedent` and the strip for each call.

Reply:
Good point